### ISSUE TITLE
SHELL_TYPE=prompt_toolkit: ImportError: cannot import name ‘create_cli’

### ISSUE THREAD

**Issue by kseistrup** [scopatz/xonsh#324]
I wanted to compare xonsh's `readline` and the `prompt_toolkit` shell types, so:

```pytb
$ SHELL_TYPE=prompt_toolkit xonsh
Traceback (most recent call last):
  File "/usr/bin/xonsh", line 3, in <module>
    main()
  File "/usr/lib/python3.4/site-packages/xonsh/main.py", line 58, in main
    shell = Shell(**shell_kwargs)
  File "/usr/lib/python3.4/site-packages/xonsh/shell.py", line 37, in __init__
    from xonsh.prompt_toolkit_shell import PromptToolkitShell
  File "/usr/lib/python3.4/site-packages/xonsh/prompt_toolkit_shell.py", line 6, in <module>
    from prompt_toolkit.shortcuts import create_cli, create_eventloop
ImportError: cannot import name 'create_cli'
```
and
```sh
$ xonsh
kseistrup@home ~ $ import xonsh as x
kseistrup@home ~ $ x.__version__
'0.1.6'
kseistrup@home ~ $ aura -As python-prompt_toolkit
aur/python-prompt_toolkit 0.43-1 (0) [installed]
    Library for building powerful interactive command lines in Python
kseistrup@home ~ $ import sys
kseistrup@home ~ $ sys.version
'3.4.3 (default, Mar 25 2015, 17:13:50) \n[GCC 4.9.2 20150304 (prerelease)]'
kseistrup@home ~ $ import prompt_toolkit.shortcuts as ptsc
kseistrup@home ~ $ [m for m in dir(ptsc) if m.startswith('create_')]
['create_asyncio_eventloop', 'create_default_application', 'create_default_layout', 'create_default_output', 'create_eventloop']
kseistrup@home ~ $ 
```

Could it be that `create_cli()` has been renamed to `create_default_application()` in the most recent incarnation of `prompt_toolkit`?

**Comment by scopatz** [comment]
Ahh yeah, I encountered this problem a couple of weeks ago.  I think that prompt-toolkit changed their API and we haven't updated.  It would be nice if this was fixed. I'll open up an issue there.

**Comment by kseistrup** [comment]
People might have different versions of `prompt_toolkit`. Assuming that `create_cli()` has indeed been renamed to `create_default_application()` the solution could be as simple as:

```py
try:
    from prompt_toolkit.shortcuts import create_cli
except ImportError:
    from prompt_toolkit.shortcuts import create_default_application as create_cli
```

I can't seem to find any reference to such a renaming in prompt_toolkit's CHANGELOG.  I'll try and clone their repository and take a look in the git log…

**Comment by scopatz** [comment]
Just opened up jonathanslenders/python-prompt-toolkit#141 to ask them what the right thing to do is.  But if you develop a fix, then great!

**Comment by kseistrup** [comment]
Pretending that `create_default_application()` is the new name for `create_cli()` threw an `AssertionError` in the `shortcuts` submodule

```pytb
Traceback (most recent call last):
  File "/usr/bin/xonsh", line 3, in <module>
    main()
  File "/usr/lib/python3.4/site-packages/xonsh/main.py", line 95, in main
    shell.cmdloop()
  File "/usr/lib/python3.4/site-packages/xonsh/prompt_toolkit_shell.py", line 102, in cmdloop
    lexer=self.lexer)
  File "/usr/lib/python3.4/site-packages/xonsh/prompt_toolkit_shell.py", line 65, in get_user_input
    key_bindings_registry=key_bindings_registry)
  File "/usr/lib/python3.4/site-packages/prompt_toolkit/shortcuts.py", line 270, in create_default_application
    extra_input_processors=extra_input_processors),
  File "/usr/lib/python3.4/site-packages/prompt_toolkit/shortcuts.py", line 125, in create_default_layout
    assert isinstance(message, text_type)
AssertionError
```

so `create_cli()` seems to be an entirely different animal (that has no reference in git log).  Let's wait and see what the author says.

**Comment by jonathanslenders** [comment]
See https://github.com/jonathanslenders/python-prompt-toolkit/issues/141 for some feedback.

There are a few things to change, but then it can be very user friendly and fast. I know documentation is still lacking, the example directory can help: https://github.com/jonathanslenders/python-prompt-toolkit/tree/master/examples . Also don't hesitate to ping me when you have other questions.

**Comment by rbrewer123** [comment]
@kseistrup, would you mind checking if #339 (now merged to master) has fixed this for you?  #360 mentions how to workaround raw color escape sequences if you see them.

**Comment by kseistrup** [comment]
This issue has been fixed. I now get a working prompt toolkit shell. Thanks!
